src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseParticipationResource.java

```diff
@@ -93,10 +98,23 @@ public ProgrammingExerciseParticipationResource(ProgrammingExerciseParticipation
         if (!programmingExerciseParticipationService.canAccessParticipation(participation.get())) {
             return forbidden();
         }
-        if (!authCheckService.isAtLeastTeachingAssistantForExercise(participation.get().getExercise())) {
+
+        // Fetch template and solution participation into exercise of participation
+        ProgrammingExercise exercise = (ProgrammingExercise) participation.get().getExercise();
+        exercise = programmingExerciseService.findWithTemplateParticipationAndSolutionParticipationById(exercise.getId());
```

Comment:
We only need the template and solution participation if the user is at least a teaching assistant or? 
In that case, it would be better if we move this into the if block in line `110`. This way we reduce unnecessary database calls. Same for the grading criteria in line `107`

Reply:
jupe, you are right. I will add an else block where I add both calls 👍